normalize-git-url.js

```diff
@@ -18,7 +18,16 @@ module.exports = function normalize (u) {
 
   var returnedUrl
   if (parsed.pathname.match(/\/?:/)) {
-    returnedUrl = u.replace(/^(?:git\+)?ssh:\/\//, '').replace(/#[^#]*$/, '')
+    if (u.match(/^git\+https?/) && parsed.pathname.match(/\/?:[^0-9]/)) {
+      returnedUrl = u.replace(/^(.*:[^:]+):(.*)/, '$1/$2')
+      console.warn('deprecated: ' + u + ' is using an' +
```

Comment:
I'm slightly concerned about spewing things out via `console.warn`, though I don't immediately have a better solution.
At the very least this is gonna corrupt and possibly be hidden by npm@3's progress bars…